lib/failure.sh

```diff
@@ -27,6 +27,20 @@ fail_invalid_package_json() {
   fi
 }
 
+fail_dot_heroku_node() {
+  if [ -f "${1:-}/.heroku/node" ]; then
+    header "Build failed"
+    warn "The directory .heroku/node could not be created
```

Comment:
Futile; do a GitHub code search for `.heroku` as a directory; it's quite common ;) But as a file it should be caught.